keep-ui/app/(keep)/workflows/[workflow_id]/layout.tsx

```diff
@@ -1,37 +1,22 @@
-"use client";
-
-import { Link } from "@/components/ui";
-import { ArrowRightIcon } from "@heroicons/react/16/solid";
-import { Icon, Subtitle } from "@tremor/react";
-import { useParams } from "next/navigation";
+import { getWorkflowWithRedirectSafe } from "@/shared/api/workflows";
+import { WorkflowBreadcrumbs } from "./workflow-breadcrumbs";
 import WorkflowDetailHeader from "./workflow-detail-header";
 
-export default function Layout({
+export default async function Layout({
   children,
   params,
 }: {
-  children: any;
+  children: React.ReactNode;
   params: { workflow_id: string };
 }) {
-  const clientParams = useParams();
+  const workflow = await getWorkflowWithRedirectSafe(params.workflow_id);
```

Comment:
we get that both in layout and page, does that mean there'll be 2 requests or it uses the same data?

Reply:
nice catch, nextjs should have dedupe it, but apparently it don't so I added react cache for it